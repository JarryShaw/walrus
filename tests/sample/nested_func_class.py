bar = 111


def foo():
    bar = 437
    lol = 0
    class A:
        @staticmethod
        def baz():
            nonlocal bar
            print(bar := 777)
        nonlocal lol
        print(lol := 999)
    A.baz()
    print(bar)
    print(lol)
    print(A.__dict__.get('lol'))


foo()
print(bar)
