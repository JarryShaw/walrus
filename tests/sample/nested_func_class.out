777
777
111
